### PR TITLE
Add option to disable duotone

### LINKED ISSUE #31764: Option to disable duotone
## What problem does this address?

There should be a way to disable the duotone UI from theme.json

## What is your proposed solution?

> I think for colors, if you set an empty palette `[]` and set custom to `false` in theme.json the control disappears entirely, so maybe something similar could work here too

### PR DESCRIPTION
**Pull request by ajlende**
## Description

Fixes #31764 

This allows for duotone to be disabled when `color.duotone` is set to `null` in theme.json as well as respecting settings when `color.custom === false` and `color.palette === null`.

## How has this been tested?

Change settings in theme.json as shown below in the screenshots section.

## Screenshots 

```json
{
	"version": 1,
	"settings": {
		"color": {
			"custom": true,
			"duotone": [],
			"palette": []
		}
	}
}
```

image

---

```json
{
	"version": 1,
	"settings": {
		"color": {
			"custom": false,
			"duotone": [],
			"palette": []
		}
	}
}
```

image

---

```json
{
	"version": 1,
	"settings": {
		"color": {
			"custom": true,
			"duotone": [],
			"palette": null
		}
	}
}
```

image

---

```json
{
	"version": 1,
	"settings": {
		"color": {
			"custom": false,
			"duotone": [],
			"palette": null
		}
	}
}
```

image

---

```json
{
	"version": 1,
	"settings": {
		"color": {
			"duotone": null
		}
	}
}
```

image

## Types of changes

New feature / Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-editor/src/hooks/duotone.js [hunk 1]
```diff
@@ -125,12 +125,14 @@ function DuotonePanel( { attributes, setAttributes } ) {
 
 	const duotonePalette = useSetting( 'color.duotone' );
 	const colorPalette = useSetting( 'color.palette' );
+	const disableCustomColors = ! useSetting( 'color.custom' );
```
Comment (gwwar) [inline packages/block-editor/src/hooks/duotone.js:128]: Could we inverse this? It reads more nicely without the double negative:
```jsx
const customColors = useSetting( 'color.custom' ); // or enableCustomColors
///...
{ customColors && (
    <CustomDuotoneBar value={ value } onChange={ onChange } />
}
```

Comment (ajlende) [reply packages/block-editor/src/hooks/duotone.js:128]: I was keeping things consistent with how [`PanelColorGradientSettings`](https://github.com/WordPress/gutenberg/blob/896ed84a811addb93fa9ab1db52f5c9a61c80641/packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js#L83) works, but keeping that consistency probably doesn't matter that much

Comment (gwwar) [reply packages/block-editor/src/hooks/duotone.js:128]: Oh interesting, this one does personally read a bit awkwardly, but It's non-blocking if we're intentionally following this pattern.

**Comment by gwwar** [comment]
Reserving this comment for manual testing:

#### Image to Cover Transform

It looks like this will throw an error when duotone is not set on an image. We can reproduce this with any combination, by clicking on "add text to image" button before duotone is applied.

Screen Shot 2021-05-19 at 9 29 36 AM

#### With duotone disabled

```
{
	"version": 1,
	"settings": {
		"color": {
			"duotone": null
		}
	}
}
```
✅ No duotone options appear on toolbar

#### Duotone enabled, No custom colors/pallets

✅ UI looks correct on toolbar

| example UI |
|-----|
| Screen Shot 2021-05-19 at 9 41 29 AM |

```
{
	"version": 1,
	"settings": {
		"color": {
			"custom": false,
			"duotone": [],
			"palette": null
		}
	}
}
```

#### Duotone enabled + custom colors

✅ UI looks correct on toolbar

| example UI |
|-----|
| Screen Shot 2021-05-19 at 9 45 21 AM |


```
{
	"version": 1,
	"settings": {
		"color": {
			"custom": true,
			"duotone": [],
			"palette": null
		}
	}
}
```
#### Duotone enabled + palette

✅ UI looks correct on toolbar

```
{
	"version": 1,
	"settings": {
		"color": {
			"custom": false,
			"duotone": [],
			"palette": []
		}
	}
}
```

This is best left for a follow up PR, but I also noticed that the first time we set a color value, this dismisses the duotone dropdown:


https://user-images.githubusercontent.com/1270189/118852983-23c35900-b888-11eb-9629-c3b3a2d296b1.mp4

#### Everything enabled

| example UI |
|-----|
| Screen Shot 2021-05-19 at 9 56 28 AM |

```
{
	"version": 1,
	"settings": {
		"color": {
			"custom": true,
			"duotone": [],
			"palette": []
		}
	}
}
```

**Comment by ajlende** [comment]
Transform from image to cover seems to have that same error even when duotone isn't disabled. I'm looking into a fix that I'll open up in another PR

EDIT: added https://github.com/WordPress/gutenberg/pull/32006

**Review by gwwar — APPROVED** [review]
Thanks for adding this quickly @ajlende! Giving tentative approval, since we have https://github.com/WordPress/gutenberg/pull/32006 

Feel free to merge when checks are green. I'd maybe rebase with trunk since the e2e failure looks a bit odd.

**Comment by youknowriad** [comment]
The test failure is already on trunk and is being investigated, so I'm merging this one.

**Comment by jasmussen** [comment]
❤️

**Comment by oandregal** [comment]
After some conversation at https://github.com/WordPress/gutenberg/pull/33280 I've prepared https://github.com/WordPress/gutenberg/pull/33295 to update the logic that handles the visibility of the duotone control.

**Pull request merged**